### DISCUSSION TITLE
de/integrations/albumentations/

### DISCUSSION THREAD

**Discussion by giscus** [ultralytics/ultralytics#17610] [Docs]
# de/integrations/albumentations/

Learn how to use Albumentations with YOLO11 to enhance data augmentation, improve model performance, and streamline your computer vision projects.

https://docs.ultralytics.com/de/integrations/albumentations/

**Comment by DP1701** [comment]
Thanks for this explanation. But if I understand correctly, not all functions of Albuminations are directly supported, or am I misunderstanding? 

The explanation says: The integration includes optimized augmentations like blur, median blur, grayscale conversion, and CLAHE with carefully tuned probabilities to enhance model performance.

So only blur, median blur, grayscale conversion and CLAHE are applied and the others cannot be individually selected and adjusted?

**Reply by glenn-jocher** [reply]
The integration with Albumentations in YOLO11 automatically applies specific augmentations like blur, median blur, grayscale conversion, and CLAHE. While these are the default transformations used to enhance model performance, you can manually customize and apply other Albumentations functions by modifying the training script to suit your specific needs. For a more comprehensive application of Albumentations' capabilities, you would need to configure these augmentations in your training pipeline separately.

**Comment by UltralyticsAssistant** [comment]
👋 Hello, thank you for your interest in Ultralytics 🚀! It seems like you're exploring Albumentations with YOLO11, which is fantastic for enhancing data augmentation and improving model performance.

For anyone new or in need of a refresher, we recommend visiting our [Docs](https://docs.ultralytics.com) where you can find extensive resources, including [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) usage examples that might answer common questions you have.

If this topic involves a 🐛 Bug Report or technical challenge, we kindly request a [minimum reproducible example](https://docs.ultralytics.com/help/minimum_reproducible_example/) to better assist with debugging any issues you might encounter.

Should your discussion pertain to a custom training ❓ Question, please share comprehensive details, such as dataset image samples and training logs. Also, have a look at our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/) to ensure optimal outcomes.

Connect with the Ultralytics community where it suits you best! For real-time discussion, join us on [Discord](https://ultralytics.com/discord) 🎧. Prefer more extensive conversations? Try our [Discourse](https://community.ultralytics.com). Or engage with the community through our [Subreddit](https://reddit.com/r/ultralytics).

## Upgrade

To ensure you have the latest capabilities and fixes, upgrade to the latest `ultralytics` package with all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) installed in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/):

```bash
pip install -U ultralytics
```

## Environments

YOLO can run smoothly in any of these verified environments, complete with all necessary dependencies like [CUDA](https://developer.nvidia.com/cuda)/[CUDNN](https://developer.nvidia.com/cudnn):

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml/badge.svg" alt="Ultralytics CI"></a>

Check if all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml?query=event%3Aschedule) tests are passing. Our CI tests run on multiple systems to verify the correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) regularly.

This is an automated response, and an Ultralytics engineer will also assist you soon. 😊
